File: frontend/src/api/series.ts
import { apiClient } from './client';

export interface Series {
  id: string;
  user_id: string;
  name: string;
  description: string | null;
  length: number;
  min_value: number;
  max_value: number;
  avg_value: number;
  created_at: string;
  updated_at: string;
}

export interface SeriesListResponse {
  items: Series[];
  total: number;
  page: number;
  page_size: number;
}

export const seriesApi = {
  upload: (file: File, name?: string) => {
    const formData = new FormData();
    formData.append('file', file);
    if (name) formData.append('name', name);
    return apiClient.post<{ series_id: string; name: string; length: number }>('/series/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
  },
  getAll: (params?: { page?: number; page_size?: number; search?: string }) =>
    apiClient.get<SeriesListResponse>('/series', { params }),
  getById: (id: string) => apiClient.get<Series>(`/series/${id}`),
  getPreview: (id: string, rows: number = 20) =>
    apiClient.get<{ headers: string[]; data: Array<{ index: number; date: string; value: number }> }>(
      `/series/${id}/preview`, { params: { rows } }
    ),
  update: (id: string, data: { name?: string; description?: string }) =>
    apiClient.patch<Series>(`/series/${id}`, data),
  delete: (id: string) => apiClient.delete(`/series/${id}`),
  getPlot: (id: string) => apiClient.get<string>(`/series/${id}/plot`),
};

File: backend/src/schemas/series.py
"""
Pydantic схемы для Series API
"""

from pydantic import BaseModel, Field
from typing import Optional, List, Any
from datetime import datetime


class SeriesCreate(BaseModel):
    """B04-01: Создание ряда (ручной ввод)"""
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    values: List[float] = Field(..., min_length=1)
    dates: Optional[List[str]] = None


class SeriesUpdate(BaseModel):
    """B04-06: Обновление метаданных ряда"""
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None


class SeriesResponse(BaseModel):
    """Ответ с информацией о ряде"""
    id: str
    user_id: str
    name: str
    description: Optional[str] = None
    length: int
    min_value: float
    max_value: float
    avg_value: float
    created_at: datetime
    updated_at: datetime


class SeriesListResponse(BaseModel):
    """Список рядов с пагинацией"""
    items: List[SeriesResponse]
    total: int
    page: int
    page_size: int


class SeriesPreviewResponse(BaseModel):
    """B04-04: Предпросмотр данных"""
    headers: List[str]
    data: List[dict]


class SeriesUploadResponse(BaseModel):
    """B04-01: Ответ после загрузки"""
    series_id: str
    name: str
    length: int
    message: str


class SeriesPlotResponse(BaseModel):
    """B04-07: График в формате base64"""
    plot: str  # base64 encoded image
    format: str = "png"
